Packs/SecurityScorecard/Integrations/SecurityScorecard/SecurityScorecard.py

```diff
@@ -0,0 +1,1078 @@
+
+    """`securityscorecard_alert_delete_command`: Delete an alert
+    See https://securityscorecard.readme.io/reference#delete_users-by-username-username-alerts-grade-alert
+    See https://securityscorecard.readme.io/reference#delete_users-by-username-username-alerts-score-alert
+
+    :type ``client``: ``Client``
+    :type `` args``: ``Dict[str, Any]``
+        ``args['alert_id']``: Alert ID, type ``String``
+        ``args['alert_type']``: Alert type. Can be either `score` or `grade`, type ``String``
+
+    :return:
+        A ``CommandResults`` object that is passed to ``return_results``
+    :rtype: ``CommandResults``
+    """
+
+    email = demisto.params().get('username').get("identifier")
+    alert_id = str(args.get("alert_id"))
+    alert_type = str(args.get("alert_type"))
+    client.delete_alert(email=email, alert_id=alert_id, alert_type=alert_type)
+
+    markdown = "{0} alert **{1}** deleted".format(str.capitalize(alert_type), alert_id)
+
+    results = CommandResults(readable_output=markdown)
+
+    return results
+
+
+def securityscorecard_alerts_list_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+
+    """`securityscorecard_alerts_list_command`: Retrieve alerts triggered in the last week
+
+    See https://securityscorecard.readme.io/reference#get_users-by-username-username-notifications-recent
+
+    :type ``client``: ``Client``
+    :type `` args``: ``Dict[str, Any]``
+        ``args['portfolio_id']``: Portfolio ID. Can be retrieved using `!securityscorecard-portfolios-list`, type ``String``
+
+    :return:
+        A ``CommandResults`` object that is passed to ``return_results``
+    :rtype: ``CommandResults``
+    """
+
+    email = demisto.params().get('username').get("identifier")
+    demisto.debug("email: {0}".format(email))
+    portfolio_id = args.get('portfolio_id')
+
+    demisto.debug("Sending request to retrieve alerts with arguments {0}".format(args))
+
+    response = client.get_alerts_last_week(email=email, portfolio_id=portfolio_id)
+
+    entries = response["entries"]
+
+    # Retrieve the alert metadata (direction, score, factor, grade_letter, score_impact)
+    alerts = []
+
+    for entry in entries:
+        # change_data is a list that includes all alert metadata that triggered the event
+        changes = entry.get("change_data")
+        for change in changes:
+            alert = {}
+            alert["id"] = entry.get("id")
+            alert["change_type"] = entry.get("change_type")
+            alert["domain"] = entry.get("domain")
+            alert["company"] = entry.get("company_name")
+            alert["created"] = entry.get("created_at")
+            alert["direction"] = change.get("direction")
+            alert["score"] = change.get("score")
+            alert["factor"] = change.get("factor")
+            alert["grade_letter"] = change.get("grade_letter")
+            alert["score_impact"] = change.get("score_impact")
+            alerts.append(alert)
```

Comment:
Are we filtering the data from the alert? what data are we leaving out? can you elaborate?

Reply:
We're not filtering data in this case.
The API returns an entry representing an alert that may include more than one changed factor data (in cases where the alert was created using multiple factors, i.e. network security and DNS health were both changed within the same alert).

What I'm doing here is creating a separate alert for each one of the factors that changed. So for example, if there was 1 alert that included network security and DNS health, I split them up into two different entries.

WDYT?

Reply:
Firstly you are giving them the same ID which is wrong. Secondly I would consult Varun on this as-well as it relate to behavior of incidents and what should be counted as incidents. How it appears in their UI?

Reply:
Modified output in commit `b2cd026d63150350c1d4b11168a0e3fab32cd721`